### ISSUE TITLE
MuJoCo for  rope drived

### ISSUE THREAD

**Issue by chenxzu** [google-deepmind/mujoco#1315]
Hi,

I'm a student and I'm trying to use MuJoCo for  rope drived.

I'm looking for some help .

I want the block to move at the same speed as the cylinder falls, and the cylinder to stop after the block stops.

But this rope will extend, causing me to be unable to achieve this effect.

In short, I hope that s2 can function as a pulley and the rope cannot be extended. I want to know how to achieve this.

Thank you so much

Here is a model which explains my question:

<details>
  <summary>minimal XML</summary>

```XML
<mujoco>
    <option gravity="0 0 -9.81" />
    <default>
        <site type="sphere" rgba="1 0 0 1" size="0.01"/>
    </default>

    <worldbody>
        <light diffuse=".5 .5 .5" pos="0 0 3" dir="0 0 -1" />
        <geom name="floor" pos="0 0 -0.5" size="0 0 1" type="plane" />


        <body name="weight" pos="0 0 .5">
            <geom type="cylinder" size="0.1 0.1" mass="1" rgba="0 .9 0 1"/>
            <joint name="w" type="slide" pos="0 0 .5" axis= "0 0 1" limited="false"/>
            <site name="s1" pos="0 0 .1"/>
        </body>

        <site name="s2" pos="0 0 1"/>

        <body name="slide" pos="0 .5 1">
            <geom type="box" size=".05 .05 .05" mass="1" rgba="0 0 .9 1"/>
            <joint name="s" type="slide" pos="0 0 0" axis= "0 1 0" limited="true" range="-0.5 0" />
            <site name="s3" pos="0 -0.05 0"/>
        </body>
    </worldbody>

    <tendon>
        <spatial name="tendon" width="0.005" damping="1" stiffness="0">
            <site site="s1"/>
            <site site="s2"/>
            <site site="s3"/>
        </spatial>
    </tendon>
</mujoco>
```

</details>

Here is a screenshot / video, illustrating my question:

https://github.com/google-deepmind/mujoco/assets/148617991/cc80154f-fb33-4df1-8a6a-266db23bb521

**Comment by chenxzu** [comment]
Hi,

I think I solved part of the problem, but now there is a new problem: how to solve the problem of rope threading through the 

mold.


Here is a model which explains my question:

```XML
<mujoco>
    <compiler autolimits="true"/>

    <default>
        <site type="sphere" rgba="1 0 0 1" size="0.01"/>
    </default>

    <worldbody>
        <light diffuse=".5 .5 .5" pos="0 0 3" dir="0 0 -1" />
        <geom name="floor" pos="0 0 -0.5" size="0 0 1" type="plane" />

        <body name="weight" pos="0 0 .5">
            <geom type="cylinder" size="0.1 0.1" mass="1" rgba="0 .9 0 1"/>
            <joint name="w" type="slide" pos="0 0 .5" axis= "0 0 1" limited="false"/>
            <site name="s1" pos="0 0 .1"/>
        </body>

        <body name="pulley" pos="0 0 1" zaxis="1 0 0" >
            <geom name="pulley" type="cylinder" size="0.1 0.01" mass="1" rgba="0 .9 0 1"/>
        </body>

        <site name="s2" pos="0 -.1 1.1"/>

        <body name="slide" pos="0 .5 1">
            <geom type="box" size=".05 .05 .05" mass="1" rgba="0 0 .9 1"/>
            <joint name="s" type="slide" pos="0 0 0" axis= "0 1 0" limited="true" range="-0.5 0" />
            <site name="s3" pos="0 -0.05 0"/>
        </body>
    </worldbody>

    <tendon>
        <spatial name="rope" range="0 1" stiffness="0">
          <site site="s1"/>
          <geom geom="pulley" sidesite="s2"/>
          <site site="s3"/>
        </spatial>
      </tendon>

</mujoco>
```


https://github.com/google-deepmind/mujoco/assets/148617991/a02fd5c9-bbb3-4c4e-8f9a-492d9b907331

**Comment by chenxzu** [comment]
I have solved this problem in another question #1317. Thank you very much for the author's answer.
